src/servicex/did_finder/rucio_adapter.py

```diff
@@ -62,29 +62,33 @@ def find_replicas(self, files, site):
                 else:
                     file_list = [{'scope': files['scope'], 'name': files['name']}]
 
-                location = {'site': site} if site else None
-
                 g_replicas = self.replica_client.list_replicas(
                     dids=file_list,
                     schemes=['root'],
-                    client_location=location)
+                    client_location=None)
 
             except Exception as eek:
                 print("\n\n\n\n\nERROR READING REPLICA ", eek)
         return g_replicas
 
     @staticmethod
-    def get_sel_path(replica, prefix):
+    def get_sel_path(replica, prefix, site):
         sel_path = None
 
         if 'pfns' not in replica:
             return None
 
+        sitename = ''
+        if site:
+            sitename = site
+
         for fpath, meta in replica['pfns'].items():
-            if not meta['type'] == 'DISK':
-                continue
-            sel_path = fpath
-            if meta['domain'] == 'lan':
+            # if meta['type'] == 'DISK' and any(stnm in fpath for stnm in sitenames):
+            if meta['type'] == 'DISK' and sitename in fpath:
+                sel_path = fpath
                 break
 
+        if not sel_path:
```

Comment:
Doesn't this create a preference for site base on sorting order, not what the user specified? Brookhaven before MWT2 no matter what the user wants? 
This would be caught by unit tests of `Rucio_ops.py` please add them and exercise the various scenarios

Reply:
In the current case we have a single site request, so this only comes into play if no replica name matches the requested site. Agreed that the full site list (and subsequent retries by site) will require a unit test suite; I propose adding it in a subsequent PR.

Reply:
Right - I forgot that we are just using one site name now